src/examples/combobox/NestedMultiselect.tsx

```diff
@@ -0,0 +1,185 @@
+    {
+      'aria-label': 'Curcubits',
+      options: [{ value: 'Cucumber' }, { value: 'Pumpkin' }, { value: 'Zucchini' }]
+    }
+  ],
+  Crucifers: [
+    { value: 'Classifications', type: 'previous' },
+    {
+      'aria-label': 'Crucifers',
+      options: [{ value: 'Broccoli' }, { value: 'Cauliflower' }, { value: 'Kale' }]
+    }
+  ],
+  Legumes: [
+    { value: 'Classifications', type: 'previous' },
+    {
+      'aria-label': 'Legumes',
+      options: [{ value: 'Bean' }, { value: 'Lentil' }, { value: 'Pea' }]
+    }
+  ],
+  Solanaceae: [
+    { value: 'Classifications', type: 'previous' },
+    {
+      'aria-label': 'Solanaceae',
+      options: [{ value: 'Bell pepper' }, { value: 'Potato' }, { value: 'Tomato' }]
+    }
+  ],
+  Umbellifers: [
+    { value: 'Classifications', type: 'previous' },
+    {
+      'aria-label': 'Umbellifers',
+      options: [{ value: 'Carrot' }, { value: 'Celery' }, { value: 'Parsnip' }]
+    }
+  ]
+};
+
+const Example = () => {
+  const [options, setOptions] = useState(OPTIONS);
+  const [state, setState] = useState<{
+    activeIndex: NonNullable<IComboboxProps['activeIndex']>;
+    inputValue: NonNullable<IComboboxProps['inputValue']>;
+    isExpanded: NonNullable<IComboboxProps['isExpanded']>;
+    selectionValue: OptionValue[];
+  }>({
+    activeIndex: -1,
+    inputValue: '',
+    isExpanded: false,
+    selectionValue: []
+  });
+
+  const handleChange: IComboboxProps['onChange'] = changes => {
+    const selectionValue = Array.isArray(changes.selectionValue) ? changes.selectionValue : [];
+    const _state = {
+      activeIndex: changes.activeIndex === undefined ? state.activeIndex : changes.activeIndex,
+      inputValue: changes.inputValue || state.inputValue,
+      isExpanded: changes.isExpanded === undefined ? state.isExpanded : changes.isExpanded,
+      selectionValue: changes.selectionValue === undefined ? state.selectionValue : selectionValue
+    };
+
+    if (selectionValue) {
+      const selectedValue = selectionValue.find(value => !state.selectionValue.includes(value));
+
+      if (selectedValue in SUB_OPTIONS) {
+        setOptions(SUB_OPTIONS[selectedValue]);
+        _state.activeIndex = 1;
+        _state.selectionValue = state.selectionValue;
+      } else if (selectedValue === 'Classifications') {
+        setOptions(OPTIONS);
+        _state.selectionValue = state.selectionValue;
+      }
+    }
+
+    setState(_state);
+  };
+
+  const renderHiddenSelectedOptions = useCallback(() => {
+    const values = options.reduce<OptionValue[]>((_values, option) => {
+      if ('options' in option) {
+        const subValues = option.options.map(subOption => subOption.value);
+
+        _values.push(...subValues);
+      }
+
+      return _values;
+    }, []);
+    const missingValues = state.selectionValue.filter(value => !values.includes(value));
+
+    return missingValues.map(value => <Option key={value as string} isHidden value={value} />);
```

Comment:
Reminder: the casting `key={value as string}` is redundant with v9.

Reply:
Yep, the v9 example is going to look cleaner ✨